src/gossip/table.zig

```diff
@@ -655,61 +658,75 @@ pub const GossipTable = struct {
+    /// location, so the indices pointing to that item need to be updated to
+    /// point to its new index.
+    ///
+    /// This is separated into a different function to isolate the state to
+    /// avoid mistakes. The new entry must be acquired from map using the
+    /// existing index. The prior entry does not point to the correct item.
+    fn accountForSwapRemove(self: *GossipTable, entry_index: usize) void {
         const table_len = self.len();
         // if (index == table_len) then it was already the last
         // element so we dont need to do anything
-        if (entry_index < table_len) {
-            // versioned data now points to the element which was swapped in and needs updating
-            const new_index_cursor = versioned_value.cursor_on_insertion;
-            const new_index_origin = versioned_value.value.id();
-
-            // update shards
-            self.shards.remove(table_len, &versioned_value.value_hash);
-            // wont fail because we just removed a value in line above
-            self.shards.insert(entry_index, &versioned_value.value_hash) catch unreachable;
-
-            // these also should not fail since there are no allocations - just changing the value
-            switch (versioned_value.value.data) {
-                .ContactInfo => {
-                    const did_remove = self.contact_infos.swapRemove(table_len);
-                    std.debug.assert(did_remove);
-                    self.contact_infos.put(entry_index, {}) catch unreachable;
-                },
-                .LegacyContactInfo => {
-                    const did_remove = self.contact_infos.swapRemove(table_len);
-                    std.debug.assert(did_remove);
-                    self.contact_infos.put(entry_index, {}) catch unreachable;
-                },
-                .Vote => {
-                    self.votes.put(new_index_cursor, entry_index) catch unreachable;
-                },
-                .EpochSlots => {
-                    self.epoch_slots.put(new_index_cursor, entry_index) catch unreachable;
-                },
-                .DuplicateShred => {
-                    self.duplicate_shreds.put(new_index_cursor, entry_index) catch unreachable;
-                },
-                else => {},
-            }
-            self.entries.put(new_index_cursor, entry_index) catch unreachable;
-
-            const new_entry_indexs = self.pubkey_to_values.getEntry(new_index_origin).?.value_ptr;
-            const did_remove = new_entry_indexs.swapRemove(table_len);
-            std.debug.assert(did_remove);
-            new_entry_indexs.put(entry_index, {}) catch unreachable;
+        std.debug.assert(entry_index <= table_len);
+        if (entry_index == table_len) return;
+
+        // replace data with newly swapped value
+        const entry = self.store.getEntryByIndex(entry_index);
+        // gossip_data now points to the element which was swapped in and needs updating
```

Comment:
think this comment is off

Reply:
e1386b151815f52f002cf13cc8f211b193609ea0